lib/logstash/outputs/newrelic.rb

```diff
@@ -64,34 +65,44 @@ def time_to_logstash_timestamp(time)
     end
   end
 
-  def encode(event_hash)
-    log_message_hash = {
-      # non-intrinsic attributes get put into 'attributes'
-      :attributes => event_hash
-    }
+  def to_NR_logs(logstash_events)
+    nr_logs = []
 
-    # intrinsic attributes go at the top level
-    if event_hash['message']
-      log_message_hash['message'] = event_hash['message']
-      log_message_hash[:attributes].delete('message')
-    end
-    if event_hash['timestamp']
-      log_message_hash['timestamp'] = event_hash['timestamp']
-      log_message_hash[:attributes].delete('timestamp')
+    logstash_events.each do |logstash_event|
```

Comment:
Should be able to do a `logstash_events.map do |logstash_event|` and not need the `nr_logs` variable separately. (`nr_logs.push(nr_log_message_hash)` at the end would instead just be `nr_log-message_hash` to return it from the block)

Reply:
Thanks! Nice one!